src/textual/widgets/_digit_display.py

```diff
@@ -0,0 +1,278 @@
+from textual.app import ComposeResult
+from textual.reactive import reactive
+from textual.widget import Widget
+from textual.widgets import Static
+
+
+_character_map: dict[str, str] = {}
+
+_VIRTUAL_SPACE = "·"
+
+# in the mappings below, we use a dot instead of spaces to make them more
+# visible, we will strip them out later
+_character_map[
+    "0"
+] = """
+┏━┓
+┃╱┃
+┗━┛
+"""
+
+_character_map[
+    "1"
+] = """
+·┓·
+ ┃·
+╺┻╸
+"""
+
+_character_map[
+    "2"
+] = """
+╺━┓
+┏━┛
+┗━╸
+"""
+
+_character_map[
+    "3"
+] = """
+╺━┓
+ ━┫
+╺━┛
+"""
+
+_character_map[
+    "4"
+] = """
+╻ ╻
+┗━┫
+  ╹
+"""
+
+_character_map[
+    "5"
+] = """
+┏━╸
+┗━┓
+╺━┛
+"""
+
+_character_map[
+    "6"
+] = """
+┏━╸
+┣━┓
+┗━┛
+"""
+
+_character_map[
+    "7"
+] = """
+╺━┓
+  ╹
+ ╹·
+"""
+
+_character_map[
+    "8"
+] = """
+┏━┓
+┣━┫
+┗━┛
+"""
+
+_character_map[
+    "9"
+] = """
+┏━┓
+┗━┫
+╺━┛
+"""
+
+_character_map[
+    " "
+] = """
+···
+···
+···
+"""
+
+
+_character_map[
+    "X"
+] = """
+╻ ╻
+·╋·
+╹ ╹
+"""
+
+_character_map[
+    "Y"
+] = """
+╻ ╻
+·┳·
+·╹·
+"""
+
+_character_map[
+    "Z"
+] = """
+╺━┓
+·▞·
+┗━╸
+"""
+
+
+_character_map[
+    "."
+] = """
+··
+··
+·•
+"""
+
+
+_character_map[
+    ","
+] = """
+··
+··
+·▞
+"""
+
+_character_map[
+    "+"
+] = """
+···
+╺╋╸
+···
+"""
+
+_character_map[
+    "-"
+] = """
+···
+╺━╸
+···
+"""
+
+_character_map[
+    "="
+] = """
+···
+╺━·
+╺━·
+"""
+
+_character_map[
+    "*"
+] = """
+···
+·✱·
+···
+"""
+
+
+_character_map[
+    "/"
+] = """
+··╻
+·▞·
+╹··
+"""
+
+_character_map[
+    "^"
+] = """
+·╻·
+▝·▘
+···
+"""
+
+
+# here we strip spaces and replace virtual spaces with spaces
+_character_map = {
+    k: v.strip().replace(_VIRTUAL_SPACE, " ") for k, v in _character_map.items()
+}
+
+
+class _SingleDigitDisplay(Static):
```

Comment:
Cat you add `get_content_width` and `get_content_height` methods. This allows "auto" dimensions pick a more optimal code path. There should be plenty of examples in other widgets...

Reply:
Hello @willmcgugan :wave: 
Were you able to have a look at this?

Reply:
If you set your widget to `width: auto;` and `height: auto` it should call `get_content_width` etc.